app/javascript/packages/rails-i18n-webpack-plugin/rails-i18n-webpack-plugin.js

```diff
@@ -0,0 +1,259 @@
+const fs = require('fs/promises');
+const { format } = require('util');
+const path = require('path');
+const YAML = require('yaml');
+const ExtractKeysWebpackPlugin = require('./extract-keys-webpack-plugin.js');
+
+/**
+ * @typedef RailsI18nWebpackPluginOptions
+ *
+ * @prop {string} template String to format using locale data object, generated using util.format.
+ * @prop {string} configPath Root path for locale configuration data.
+ * @prop {string} defaultLocale Default locale to use if string data is missing for desired locale.
+ * @prop {string} missingStringValue Value to use if string data is missing for both desired and
```

Comment:
> could we make this a function that takes in the key? then we could do something like Rails does where the missing string value is `missing translation: x.y.z`

Yeah, I like that idea. Maybe even take it a step further and throw an error in said function callback if a string is missing, so we can catch them in CI?

Reply:
so in development it would return the string, but in test/CI we'd override to throw? makes sense to me

Reply:
> so in development it would return the string, but in test/CI we'd override to throw? makes sense to me

I figured to just throw all the time, but on reflection I suppose that might be rather disruptive to development if we expect non-English values to be added later.